### PR TITLE
Improve plugins in `Alignment/HIPAlignmentAlgorithm` and add unit tests

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

The main goal of this PR is refresh the class `LhcTrackAnalyzer.cc` to add more branches relevant to study track quantities.
In addition:
   * added a `fillDesciptions` method to `HIPTwoBodyDecayAnalyzer`
   * cleaned up unused methods
   * added `catch2`-based unit tests to test these analyzers. 

#### PR validation:

`scram b runtests_testHIPAnalyzers` runs fine.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, might be backported to `CMSSW_14_0_X` for data-taking purposes.

### CODE REVIEW

File: Alignment/HIPAlignmentAlgorithm/test/testHIPAnalyzers.cc [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  // the following part is commented because the
+  // HIPTwoBodyDecayAnalyzer crashes on
+  // No "TransientTrackRecord" record found in the EventSetup.
+
+  /* 
+  SECTION("No event data") {
+   edm::test::TestProcessor tester(config);
+   REQUIRE_NOTHROW(tester.test());
```
Comment (mmusich) [inline Alignment/HIPAlignmentAlgorithm/test/testHIPAnalyzers.cc:33]: alas this fails with :

```
----- Begin Fatal Exception 19-Jun-2024 13:45:55 CEST-----------------------
An exception of category 'NoRecord' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path '_test_endpath'
   [2] Calling method for module HIPTwoBodyDecayAnalyzer/'trackAnalyzer'
Exception Message:
No "TransientTrackRecord" record found in the EventSetup.

 The Record is delivered by an ESSource or ESProducer but there is no valid IOV for the synchronization value.
 Please check
   a) if the synchronization value is reasonable and report to the hypernews if it is not.
   b) else check that all ESSources have been properly configured.
----- End Fatal Exception -------------------------------------------------
```

Despite I explicitly add the `ESProducer` in the `catch2` configuration:

https://github.com/cms-sw/cmssw/blob/0eec2cb3f17cb5a9f0cdddc319e8a5a7c84046b1/Alignment/HIPAlignmentAlgorithm/test/testHIPAnalyzers.cc#L57-L58

any help / hints are welcome.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by perrotta** [comment]
@mmusich this PR seems good to go to me.
Should I sign it for AlCa, or would you prefer waiting for the resolution of https://github.com/cms-sw/cmssw/pull/45268#discussion_r1646020617 before proceeding?

**Comment by mmusich** [comment]
> Should I sign it for AlCa, or would you prefer waiting for the resolution of https://github.com/cms-sw/cmssw/pull/45268#discussion_r1646020617 before proceeding?

I guess it can be signed as it is, though I would be happy to implement improvement anyone has comments on that. We can also come back to this later.

**Comment by perrotta** [comment]
+alca

**Comment by antoniovilela** [comment]
+1

**Pull request merged**